maintenance/log/sink.go

```diff
@@ -126,3 +153,70 @@ func (s *Sink) Write(b []byte) (int, error) {
 
 	return s.output.Write(b)
 }
+
+func (s *Sink) initialize() {
```

Comment:
I think we should not initialize the buffer with a fixed size before, this will be very costly for every HTTP request. I prefer the logic from before where the append is growing the buffer on demand. But this buffer should be kept at either defaultSinkSize or CustomSize. So I would prefer to keep a max value.

Reply:
It's not especially costly. Each uninitialized string has a memory use of 2*sizeof(int). That's something I believe we can afford. I prefer this to the additional time added by an append() call and the resulting complication in the implementation.